demos/eigenvalues_QG_basinmodes/qgbasinmodes.py.rst

```diff
@@ -214,7 +223,7 @@ solver::
 
 and we gather the final eigenfunctions ::
 
-   eigenmodes_real.vector()[:], eigenmodes_imag.vector()[:] = vr, vi
+   eigenmodes_real.vector()[ISin.array], eigenmodes_imag.vector()[ISin.array] = vr, vi
```

Comment:
You should also insert the Dirichlet conditions into these eigenvectors (`bc.apply`)

Reply:
Is this necessary, given that `bc = DirichletBC(Vcg, 0.0, "on_boundary")`?